soc/arm/silabs_exx32/Kconfig

```diff
@@ -27,7 +27,32 @@ config SOC_PART_NUMBER
 	  that you should not set directly. The part number selection choice defines
 	  the default value for this string.
 
-config HAS_EMU
+choice SOC_GECKO_DCDC_MODE
+	prompt "DC/DC mode"
+	depends on SOC_GECKO_EMU_DCDC
+	help
+	  Select power configuration mode of the on chip DC/DC converter.
+
+	config SOC_GECKO_EMU_DCDC_MODE_UNCONFIGURED
+		bool "Initial / Uconfigured"
+
+	config SOC_GECKO_EMU_DCDC_MODE_ON
+		bool "DC/DC On"
+
+	config SOC_GECKO_EMU_DCDC_MODE_OFF
+		bool "DC/DC Off"
+
+	config SOC_GECKO_EMU_DCDC_MODE_BYPASS
+		bool "Bypass"
+endchoice
+
+config SOC_GECKO_EMU_DCDC
+	bool "Enable SOC DC/DC regulator"
+	select SOC_GECKO_EMU
+	help
+	  Enable the DC/DC for the SoC.
```

Comment:
Better 'Enable the on chip DC/DC regulator' since 'DC/DC for the SoC' may also mean an external DC/DC. Or, since this text is not really different from the name of the option, we could get rid of the help text all together.